Source/WebCore/page/playstation/ResourceUsageThreadPlayStation.cpp

```diff
@@ -55,17 +57,33 @@ void ResourceUsageThread::platformCollectCPUData(JSC::VM*, ResourceUsageData& da
 
 void ResourceUsageThread::platformCollectMemoryData(JSC::VM* vm, ResourceUsageData& data)
 {
-    // FIXME: Calculate total dirty memory size for ResourceUsageOverlay
-    data.totalDirtySize = 0;
+    auto& categories = data.categories;
+
+    data.totalDirtySize = bmalloc::api::memoryFootprint();
 
     size_t currentGCHeapCapacity = vm->heap.blockBytesAllocated();
     size_t currentGCOwnedExtra = vm->heap.extraMemorySize();
     size_t currentGCOwnedExternal = vm->heap.externalMemorySize();
     RELEASE_ASSERT(currentGCOwnedExternal <= currentGCOwnedExtra);
 
-    data.categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
-    data.categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
-    data.categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+    categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
+    categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
+    categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+
+    if (isFastMallocEnabled()) {
+        showmap::Result<256> result;
+        result.collect();
+        if (auto* entry = result.entry("SceNKFastMalloc"))
+            categories[MemoryCategory::bmalloc].dirtySize = entry->effectiveRss();
+
+        // TODO: collect dirty size of "MemoryCategory::Images" and "MemoryCategory::Layers"
+        if (data.totalDirtySize > categories[MemoryCategory::bmalloc].dirtySize)
+            categories[MemoryCategory::Other].dirtySize = data.totalDirtySize - categories[MemoryCategory::bmalloc].dirtySize;
+    }
+
+    auto& mallocBucket = isFastMallocEnabled()? data.categories[MemoryCategory::bmalloc]: data.categories[MemoryCategory::Other];
+    mallocBucket.dirtySize -= currentGCHeapCapacity;
+    mallocBucket.dirtySize -= std::min(mallocBucket.dirtySize, categories[MemoryCategory::GCOwned].dirtySize);
```

Comment:
Please put this logic inside if and else  even if some logic was duplicated to simplify overall code.